src/Validate.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {Contradictions, Groupings, SquareAddress, Values} from "./Types";
+
+const ROWS =   [[0 , 1 , 2 , 3 , 4 , 5 , 6 , 7 , 8],
+                [9 , 10, 11, 12, 13, 14, 15, 16, 17],
+                [18, 19, 20, 21, 22, 23, 24, 25, 26],
+                [27, 28, 29, 30, 31, 32, 33, 34, 35],
+                [36, 37, 38, 39, 40, 41, 42, 43, 44],
+                [45, 46, 47, 48, 49, 50, 51, 52, 53],
+                [54, 55, 56, 57, 58, 59, 60, 61, 62],
+                [63, 64, 65, 66, 67, 68, 69, 70, 71],
+                [72, 73, 74, 75, 76, 77, 78, 79, 80]];
+
+const COLUMNS = [[0, 9 , 18, 27, 36, 45, 54, 63, 72],
+                 [1, 10, 19, 28, 37, 46, 55, 64, 73],
+                 [2, 11, 20, 29, 38, 47, 56, 65, 74],
+                 [3, 12, 21, 30, 39, 48, 57, 66, 75],
+                 [4, 13, 22, 31, 40, 49, 58, 67, 76],
+                 [5, 14, 23, 32, 41, 50, 59, 68, 77],
+                 [6, 15, 24, 33, 42, 51, 60, 69, 78],
+                 [7, 16, 25, 34, 43, 52, 61, 70, 79],
+                 [8, 17, 26, 35, 44, 53, 62, 71, 80]];
+
+function checkGroup(values: Values, addresses: SquareAddress[]): Contradictions {
```

Comment:
Though this logic is correct, I'd recommend using lodash's [groupBy](https://lodash.com/docs/#groupBy) for clarity.

```js
const duplicateAddresses = _.chain(addresses)
  .map((address) => {
    const value = values.get(address);
    return [value, address];
  })
  .filter(([value, _]) => value !== undefined) // Ignore empty squares
  .groupBy(relevantValues, ([value, _]) => value) // Group by square value
  .filter(([_, addresses]) => address.length > 1) // Ignore values with only one address
  .map(([_, addresses]) => addresses) // Select the duplicate addresses
  .flatten()
  .values();

return new Set<SquareAddress>(duplicateAddresses);
```